sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImplTest.java

```diff
@@ -89,28 +91,35 @@
 import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
-import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
 
 /** Tests for {@link BigQueryServicesImpl}. */
 @RunWith(JUnit4.class)
 public class BigQueryServicesImplTest {
   @Rule public ExpectedException thrown = ExpectedException.none();
   @Rule public ExpectedLogs expectedLogs = ExpectedLogs.none(BigQueryServicesImpl.class);
-  @Mock private LowLevelHttpResponse response;
+  // A test can make mock responses through setupMockResponses
+  private LowLevelHttpResponse[] responses;
```

Comment:
Previously one mock response object was reused throughout multiple HTTP responses. This does not work any more as newer google-http-client calls `getStatusCode()` multiple times per response.

Now this test prepares different mock response objects for different responses.